packages/wallet/src/util.ts

```diff
@@ -1 +1,151 @@
+        } else {
+          return Buffer.from(parsedAddress.to_bytes()).toString('hex');
+        }
+      } catch (error) {
+        logger.error(error);
+        throw new ApiError(500, error);
+      }
+    },
+    getNetworkId: async (): Promise<number> => {
+      logger.debug('getting networkId');
+      return Promise.resolve(props.keyAgent.networkId);
+    },
+    getRewardAddresses: async (): Promise<Cbor[]> => {
+      logger.debug('getting reward addresses');
+      try {
+        const [{ rewardAccount }] = await firstValueFrom(wallet.addresses$);
+        const parsedAddress = parseCslAddress(rewardAccount.toString());
+
+        if (!rewardAccount || !parsedAddress) {
+          throw new ApiError(500, 'could not get reward address');
+        } else {
+          return [Buffer.from(parsedAddress.to_bytes()).toString('hex')];
+        }
+      } catch (error) {
+        logger.error(error);
+        throw new ApiError(500, error);
+      }
+    },
+    getUnusedAddresses: async (): Promise<Cbor[]> => {
+      logger.debug('getting unused addresses');
+      return Promise.resolve([]);
+    },
+    getUsedAddresses: async (_paginate?: Paginate): Promise<Cbor[]> => {
+      logger.debug('getting changeAddress');
+
+      const [{ address }] = await firstValueFrom(wallet.addresses$);
+      const parsedAddress = parseCslAddress(address.toString());
+
+      if (!address || !parsedAddress) {
+        throw new ApiError(500, 'could not get used addresses');
+      } else {
+        return [Buffer.from(parsedAddress.to_bytes()).toString('hex')];
+      }
+    },
+    getUtxos: async (amount?: Cbor, paginate?: Paginate): Promise<Cardano.Utxo[] | undefined> => {
+      let utxos = await firstValueFrom(wallet.utxo.available$);
+
+      if (amount) {
+        try {
+          const filterAmount = CSL.Value.from_bytes(Buffer.from(amount, 'hex'));
+          /**
+           * Getting UTxOs to meet a required amount is a complex operation, which is handled by input selection capabilities.
+           * By initializing a transaction we're able to utilise the internal configuration and algorithm to make this selection,
+           * using a wallet address to satisfy the interface only.
+           */
+          const { inputSelection } = await wallet.initializeTx({
+            outputs: new Set([{ address: wallet.addresses$.value![0].address, value: cslToCore.value(filterAmount) }])
+          });
+
+          utxos = [...inputSelection.inputs];
+        } catch (error) {
+          logger.debug(error);
+          throw new ApiError(400, error);
+        }
+      } else if (paginate) {
+        utxos = utxos.slice(paginate.page * paginate.limit, paginate.page * paginate.limit + paginate.limit);
+      }
+
+      return Promise.resolve(utxos);
+    },
+    signData: async (addr: Cardano.Address, payload: Bytes): Promise<Bytes> => {
+      logger.debug('signData');
+      const { signature } = await cip30signData({
```

Comment:
The return type of this function should be [DataSignature](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0030#datasignature) object, which also includes the key.
```suggestion
      return cip30signData({
```